File: matrixElem/matrixElem.py
"""
Given matrix, a rectangular matrix of integers, where each value represents the cost of the room, your task is to return the total sum of all rooms that are suitable for the CodeBots (ie: add up all the values that don't appear below a 0).

Example

    For

    matrix = [[0, 1, 1, 2], 
              [0, 5, 0, 0], 
              [2, 0, 3, 3]]

    the output should be
    matrixElementsSum(matrix) = 9.
"""

yungMatrix = [[0, 1, 1, 2], 
              [0, 5, 0, 0], 
              [2, 0, 3, 3]]

def matrixElementSum(matrix):
    myDict = {}
    counter = 0
    for x in matrix:
        #print(x)
        for y in range(len(x)):
            #print(x)
            #print(y)
            print(x[y])
            if y in myDict:
                #print('foo')
                continue
            if x[y] == 0:
                myDict[y] = y
            else:
                counter += x[y]
    print(counter)
    print(myDict)

matrixElementSum(yungMatrix)

File: matrixElem/matrixElem.js
function matrixElementsSum(matrix) {
    let counter = new Map();
    let bigTippin = 0;
    //loop through matrix by row
    for (let i=0; i<matrix.length; i++) {
	//loop through matrix by item
	//onsole.log(matrix[i], ' foo');
	//console.log(matrix[i].length)
	
	for (let y=0; y<matrix[i].length; y++) {
	    
	    
	    if (matrix[i][y] === 0){
		counter.set(y, y);
	    }
	    if(counter.has(y)) {
	    } else {
		bigTippin+= matrix[i][y];
		console.log(matrix[i][y], ` bar ${y}`)
	    }
	}
	
    }
    //console.log(counter);
    //console.log(counter.has(1))
    //console.log(bigTippin);
    return bigTippin;

}
